packages/web/src/app/NewestProducts.component.tsx

```diff
@@ -1,16 +1,33 @@
+"use client";
+
 import Image from "next/image";
 import Link from "next/link";
-import { rubberDuckData } from "@/data/data";
-import RubberDuck from "@/lib/model/rubberduck/Rubberduck.type";
+import { z } from "zod";
+import RubberDuck, { RubberDuckSchema } from "@/lib/model/rubberduck/Rubberduck.type";
 import { Utils } from "@/lib/utils/mod";
+import { useState, useEffect } from "react";
 
-const products: RubberDuck[] = rubberDuckData;
+const DucksArraySchema = z.array(RubberDuckSchema);
 
 /**
  * @component NewestProducts
  * @description Displays a list of the newest products in a grid layout.
  */
 export default function NewestProducts() {
+    const [products, setProducts] = useState<RubberDuck[]>([]);
+
+    useEffect(() => {
+        fetch("/api/ducks")
+            .then((res) => res.json())
+            .then((data) => {
+                setProducts(DucksArraySchema.parse(data));
+            })
+            .catch((error) => {
+                console.error("Error fetching products:", error);
+                setProducts([]); // Set to empty array on error
+            });
```

Comment:
Since you chose to do it the next.js way in `packages/web/src/app/items/[id]/page.tsx`, why is this an api call through the javascript fetch api instead of the typical next.js function?